packages/contracts/contracts/registry/RegistryUtils.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.10;
+
+/// @notice Checks if the name is either 0-9, a-z or a dash (-).
+/// @param subDomain The name of the DAO.
+/// @return `true` if the name is valid or `false` if at least one char is invalid.
+/// @dev Aborts on the first invalid char found.
+function isSubdomainValid(string calldata subDomain) pure returns (bool) {
+    bytes calldata nameBytes = bytes(subDomain);
+    uint256 nameLength = nameBytes.length;
+    for (uint256 i; i < nameLength; i++) {
+        uint8 char = uint8(nameBytes[i]);
```

Comment:
Isn't it possible to compare below, even without casting to uint8? (in case it saves some gas)

Reply:
nope you cannot use `>` and `<` with `bytes`. Only `uint` can be used for these comparisions